### PR TITLE
Add rom stuff

### PR DESCRIPTION
**Pull request by devsnek**
Exposes rom funcs and rom data.

### CODE REVIEW

File: rp2040-hal/src/rom_data.rs [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+
+/// A bootrom function table code.
+pub type RomFnTableCode = [u8; 2];
+
+/// This function searches for (table)
+type RomTableLookupFn<T> = unsafe extern "C" fn(*const u16, u32) -> T;
+
+/// The following addresses are described at `2.8.3. Bootrom Contents`
+/// Pointer to the lookup table function supplied by the rom.
+const ROM_TABLE_LOOKUP_PTR: *const u16 = 0x18 as _;
```
Comment (devsnek) [inline rp2040-hal/src/rom_data.rs:11]: couldn't you just type this const as a function instead of doing a transmute?

Comment (AsafFisher) [reply rp2040-hal/src/rom_data.rs:11]: A function that returns `T`?

Comment (devsnek) [reply rp2040-hal/src/rom_data.rs:11]: oh right its generic. nvm.

File: rp2040-hal/src/rom_data.rs [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+const FUNC_TABLE: *const *const u16 = 0x14 as _;
+
+/// Pointer to the public data lookup table.
+const DATA_TABLE: *const *const u16 = 0x16 as _;
+
+/// Retrive rom content from a table using a code.
+fn rom_table_lookup<T>(table: *const *const u16, tag: RomFnTableCode) -> T {
+    unsafe {
+        let _rom_table_lookup: RomTableLookupFn<T> = core::mem::transmute(ROM_TABLE_LOOKUP_PTR);
+        _rom_table_lookup(*table, u16::from_le_bytes(tag) as u32)
```
Comment (devsnek) [inline rp2040-hal/src/rom_data.rs:23]:
```suggestion
        let rom_table_lookup: RomTableLookupFn<T> = core::mem::transmute(ROM_TABLE_LOOKUP_PTR);
        rom_table_lookup(*table, u16::from_le_bytes(tag) as u32)
```

Comment (AsafFisher) [reply rp2040-hal/src/rom_data.rs]: Done lets put it in

**Pull request merged**